Realdom.php

```diff
@@ -201,7 +201,7 @@
                     next($this->_arguments);
                 }
 
-                while (list($name, $default) = each($this->_arguments)) {
+                foreach ($this->_arguments as $name => $default) {
```

Comment:
It might be not so straight…
See these lines
- https://github.com/hoaproject/Realdom/blob/master/Realdom.php#L172
- https://github.com/hoaproject/Realdom/blob/master/Realdom.php#L200

foreach() restart from the beginning while each just continues from current pointer position.
The usage of each() might have been purposely used here (without reading precisely nobody will know except @Hywan )

So I suggest two solutions
- check there is no bugs with this change
- use a nice combinaison of `while` `value($array)`, `key($array)` and `next($array)`

Reply:
You are right, it seems too easy to be fine 😄. I'll try to take a deeper look at the code or let @Hywan to tell use about the logic here.

Reply:
`each` return keypair and then move the cursor.
I guess we should replace this code by a do while with next like @guiled suggest.
```php
do {
  $name = key($this->_arguments);
  $default = current($this->_arguments);
} while(next($this->_arguments));
```

Reply:
I've updated the loop with a `do {} while();`. Now, it's compliant with the `each` logic.